### PR TITLE
[#9] [Integrate] As a user, I can see the home header

### PR DESCRIPTION
**Pull request by hoangnguyen92dn**
Close https://github.com/hoangnguyen92dn/survey-flutter-ic/issues/9

## What happened 👀

Once logged in, users can navigate to the home screen and see the home header.

## Insight 📝

- Bind profile avatar from API response
- Add UnitTest

### CODE REVIEW

File: lib/ui/home/home_header.dart [hunk 1]
```diff
@@ -20,8 +30,7 @@ class HomeHeader extends StatelessWidget {
                   crossAxisAlignment: CrossAxisAlignment.start,
                   children: [
                     Text(
-                      // TODO: Integrate with VM to get the current date
-                      'Monday, June 15'.toUpperCase(),
+                      widget.date,
```
Comment (Wadeewee) [inline lib/ui/home/home_header.dart:33]:
```suggestion
                      widget.date.toUpperCase(),
```

File: lib/ui/home/home_screen.dart [hunk 2]
```diff
@@ -1,14 +1,61 @@
...
+            ref.read(_loadingStateProvider.notifier).state = true;
+          },
+          error: (message) => {
+                ref.read(_loadingStateProvider.notifier).state = false,
+                showToastMessage(message)
+              },
+          orElse: () => {});
+    });
+    return AnnotatedRegion<SystemUiOverlayStyle>(
+      value: SystemUiOverlayStyle.light,
```
Comment (Wadeewee) [inline lib/ui/home/home_screen.dart:47]: Do we need a `AnnotatedRegion` here 🤔 
```suggestion

```

File: test/ui/home/home_view_model_test.dart [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    late MockGetProfileUseCase mockGetProfileUseCase;
+    late ProviderContainer container;
+
+    setUp(() {
+      mockGetProfileUseCase = MockGetProfileUseCase();
+
+      container = ProviderContainer(overrides: [
+        homeViewModelProvider
+            .overrideWith((ref) => HomeViewModel(mockGetProfileUseCase))
+      ]);
```
Comment (Wadeewee) [inline test/ui/home/home_view_model_test.dart:23]: Let's create `late HomeViewModel viewModel;`
```
viewModel = container.read(homeViewModelProvider.notifier);
```

File: lib/ui/home/home_screen.dart (hunk 2 above)
Comment (hoangnguyen92dn) [reply lib/ui/home/home_screen.dart:47]: @Wadeewee Seems rebase mistake, since you already added on the `main`, so we could get rid of this.

**Review by Wadeewee — APPROVED** [review]

**Pull request merged**